sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py

```diff
@@ -0,0 +1,114 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import TYPE_CHECKING
+
+from azure.core.tracing.decorator import distributed_trace
+
+from ._models import RoleAssignment, RoleDefinition
+from ._internal import KeyVaultClientBase
+
+if TYPE_CHECKING:
+    from typing import Any, Union
+    from uuid import UUID
+    from azure.core.paging import ItemPaged
+
+
+class KeyVaultAccessControlClient(KeyVaultClientBase):
+    """Manages role-based access to Azure Key Vault.
+
+    :param str vault_url: URL of the vault the client will manage. This is also called the vault's "DNS Name".
+    :param credential: an object which can provide an access token for the vault, such as a credential from
+        :mod:`azure.identity`
+    """
+
+    # pylint:disable=protected-access
+
+    @distributed_trace
+    def create_role_assignment(self, role_scope, role_assignment_name, role_definition_id, principal_id, **kwargs):
+        # type: (str, Union[str, UUID], str, str, **Any) -> RoleAssignment
+        """Create a role assignment.
+
+        :param str role_scope: scope the role assignment will apply over
```

Comment:
For the other languages we are using a custom type to provide some context for the valid values that can be passed here ("/", "/keys", "/keys/_some specific Key resource Id_"). Is there a pythonic way to accomplish the same thing here?

Reply:
There's nothing I know of in the standard library quite like .NET's `Uri`. Typical Python just passes strings around. I think validating the argument with a regex is the best we could do here. I think it's unlikely but not inconceivable we'd do that; for today I'll punt 🏈

Reply:
Makes sense - perhaps just some hints in the comment text would be enough.